storage/src/mmr/verification.rs

```diff
@@ -234,11 +235,13 @@ impl<H: CHasher> Proof<H> {
                 siblings.sort_by(|a, b| b.0.cmp(&a.0));
             }
         }
-        hashes.extend(
+        let sibling_hashes: Vec<H::Digest> = futures::future::join_all(
```

Comment:
may want to do `use futures::future::join_all;` to just do `join_all`

Reply:
(side note: super cool we are thinking about this from the start)

Reply:
done

Reply:
Updated this to also include the peak hashes in this async bulk fetch